Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Revolver.cs

```diff
@@ -393,10 +395,12 @@ private void OnRevolverTakeAmmo(EntityUid uid, RevolverAmmoProviderComponent com
                 args.Ammo.Add((spawned, EnsureComp<AmmoComponent>(spawned)));
 
                 if (cartridge.DeleteOnSpawn)
+                    component.AmmoSlots[index] = null;
                     component.Chambers[index] = null;
```

Comment:
Only the first statement is inside the if since it is missing curly braces. This and line 403 should also be upstreamed.